File: windows-build-20260602-stack-box-direct-real-bepu-fix5/cooked/code/gameplay/MenuPanelDefinition.cpp
#ifdef DrawText
#undef DrawText
#endif
#include "MenuPanelDefinition.hpp"
#include "runtime/native_string.hpp"
#include "runtime/array.hpp"
#include "MenuItemDefinition.hpp"
#include "runtime/native_exceptions.hpp"
#include "MenuPanelDefinition.hpp"
#include "runtime/array.hpp"
#include "runtime/native_exceptions.hpp"
#include "runtime/native_string.hpp"

const std::string& MenuPanelDefinition::get_Heading()
{
return this->Heading;
}

Array<::MenuItemDefinition*>* MenuPanelDefinition::get_Items()
{
return this->Items;
}

const std::string& MenuPanelDefinition::get_PanelId()
{
return this->PanelId;
}

int32_t MenuPanelDefinition::get_VisibleItemCount()
{
return this->VisibleItemCount;
}

MenuPanelDefinition::MenuPanelDefinition(std::string panelId, std::string heading, int32_t visibleItemCount, Array<::MenuItemDefinition*>* items) : Heading(), Items(), PanelId(), VisibleItemCount(0)
{
    if (String::IsNullOrWhiteSpace(panelId))
    {
throw ([&]() {
auto __ctor_arg_00000048 = "Panel id must be provided.";
auto __ctor_arg_00000049 = "panelId";
return new ArgumentException(__ctor_arg_00000048, __ctor_arg_00000049);
})();
    }
    if (String::IsNullOrWhiteSpace(heading))
    {
throw ([&]() {
auto __ctor_arg_0000004A = "Panel heading must be provided.";
auto __ctor_arg_0000004B = "heading";
return new ArgumentException(__ctor_arg_0000004A, __ctor_arg_0000004B);
})();
    }
    if (visibleItemCount < 1)
    {
throw ([&]() {
auto __ctor_arg_0000004C = "visibleItemCount";
auto __ctor_arg_0000004D = "Visible item count must be at least one.";
return new ArgumentOutOfRangeException(__ctor_arg_0000004C, __ctor_arg_0000004D);
})();
    }
    if (items == nullptr)
    {
throw new ArgumentNullException("items");
    }
    if (items->get_Length() == 0)
    {
throw new InvalidOperationException("Menu panels must contain at least one item.");
    }
this->PanelId = panelId;
this->Heading = heading;
this->VisibleItemCount = visibleItemCount;
this->Items = items;
}

File: assets/codebase/menu.authoring/MenuPanelDefinition.cs
namespace city.menu {
    /// <summary>
    /// Describes one logical city demo-disc menu screen hosted inside the shared menu scene.
    /// </summary>
    public sealed class MenuPanelDefinition {
        /// <summary>
        /// Initializes one city menu panel definition.
        /// </summary>
        /// <param name="panelId">Stable panel identifier used for navigation.</param>
        /// <param name="heading">Primary heading rendered when the panel is active.</param>
        /// <param name="visibleItemCount">Maximum number of visible rows before the panel scrolls.</param>
        /// <param name="items">Selectable items rendered by the panel.</param>
        public MenuPanelDefinition(string panelId, string heading, int visibleItemCount, [NativeTakesOwnership] MenuItemDefinition[] items) {
            if (string.IsNullOrWhiteSpace(panelId)) {
                throw new ArgumentException("Panel id must be provided.", nameof(panelId));
            }
            if (string.IsNullOrWhiteSpace(heading)) {
                throw new ArgumentException("Panel heading must be provided.", nameof(heading));
            }
            if (visibleItemCount < 1) {
                throw new ArgumentOutOfRangeException(nameof(visibleItemCount), "Visible item count must be at least one.");
            }
            if (items == null) {
                throw new ArgumentNullException(nameof(items));
            }
            if (items.Length == 0) {
                throw new InvalidOperationException("Menu panels must contain at least one item.");
            }

            PanelId = panelId;
            Heading = heading;
            VisibleItemCount = visibleItemCount;
            Items = items;
        }

        /// <summary>
        /// Gets the stable panel identifier.
        /// </summary>
        public string PanelId { get; }

        /// <summary>
        /// Gets the heading rendered for the panel.
        /// </summary>
        public string Heading { get; }

        /// <summary>
        /// Gets the maximum number of visible rows before the panel scrolls.
        /// </summary>
        public int VisibleItemCount { get; }

        /// <summary>
        /// Gets the selectable items rendered by the panel.
        /// </summary>
        public MenuItemDefinition[] Items { get; }
    }
}
